pkg/view/csrf.go

```diff
@@ -0,0 +1,69 @@
+package view
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"net/http"
+)
+
+func generateCSRFToken() (string, error) {
```

Comment:
Is there a technical reason not to use this: https://github.com/gorilla/csrf/tree/main

You're already importing it in `go.mod`. If there is a specific technical deficiency with that library then I'm open to hearing but otherwise there's no need for us to maintain our own implementation of something which is not core to the product.

Reply:
I swear I ran into [this bug](https://github.com/gobuffalo/buffalo/issues/1067), which I have used this library before. But now that I am putting together a minimal reproduction case, I am not getting the bug to trigger. 

The problem I have seen is that the library requires `SESSION_SECRET` set in the environment, which is an undocumented environmental dependency. This design pattern leads to many head-scratching problems both in local dev and production, and I avoid code that has it.

I am also perusing the libraries' source code to try to find the source of the bug, but my static analysis eyes can't find it.

Let me try the library again and see if I get that problem...

Reply:
alright, odd, I wasn't able to repro. I agree it should be gorilla/csrf, but be warned that I ran into that bug again somehow.